bamboo/common_python/tools.py

```diff
@@ -0,0 +1,207 @@
+                        num_processes/num_nodes)
+            if partition != None:
+                # q => Submits the job to one of the specified queues.
+                option_partition = ' -q %s' % partition
+            if time_limit != None:
+                # W => Sets the runtime limit of the job.
+                option_time_limit = ' -W %d' % time_limit
+            command_allocate = '%s%s%s%s%s%s%s%s' % (
+                command_allocate, option_exclusive, option_group,
+                option_interactive, option_num_processes, option_partition,
+                option_processes_per_node, option_time_limit)
+
+        # Create run command
+        if command_allocate == '':
+            command_run = 'mpirun'
+        else:
+            command_run = ' mpirun'
+        option_num_processes = ''
+        option_processes_per_node = ''
+        if num_processes != None:
+            # -np => Run this many copies of the program on the given nodes.
+            option_num_processes = ' -np %d' % num_processes
+            if (num_nodes != None) and (num_nodes != 0):
+                option_processes_per_node = ' -N %d' % int(
+                    num_processes/num_nodes)
+        command_run = '%s%s%s' % (
+            command_run, option_num_processes, option_processes_per_node)
+        
+    else:
+        raise Exception('Unsupported Scheduler %s' % scheduler)
+
+    # Create LBANN command
+    option_data_filedir = ''
+    option_data_reader = ''
+    option_data_reader_percent = ''
+    option_exit_after_setup = ''
+    option_mini_batch_size = ''
+    option_model = ''
+    option_num_epochs = ''
+    option_optimizer = ''
+    option_processes_per_model = ''
+    if model_path != None:
+        # If model_folder and/or model_name are set, an exception will be
+        # raised later.
+        option_model = ' --model=%s' % model_path
+    if data_reader_path != None:
+        # If data_reader_name is set, an exception will be raised later.
+        option_data_reader_name = ' --reader=%s' % data_reader_path
+    if optimizer_path != None:
+        # If optimizer_name is set, an exception will be raised later.
+        option_optimizer_name = ' --optimizer=%s' % optimizer_path
+    if dir_name != None:
+        if model_path != None:
+            if (model_folder != None) and (model_name != None):
+                raise Exception('Invalid usage: model_path is set but so are model folder and model_name')
+            elif model_folder != None:
+                raise Exception(
+                    'Invalid usage: model_path is set but so is model folder')
+            elif model_name != None:
+                raise Exception(
+                    'Invalid usage: model_path is set but so is model_name')
+        else:
+            if (model_folder != None) and (model_name != None):
+                option_model = ' --model=%s/model_zoo/models/%s/model_%s.prototext' % (dir_name, model_folder, model_name)
```

Comment:
prototext models that are in the 'tests' subdir are incorrect here. Generates model_zoo/models/tests/mnist_partitioned_io.prototext when it should generate model_zoo/tests/mnist_partitioned_io.prototext